### PR TITLE
Fix SignUp issue

### LINKED ISSUE #1291: amp signup: a not validated user is still created on mailer error
At signup with option validation by email, if the SendGrid key doesn't exist for instance, the user is still created with a status "Not verified", but then there is not way to remove or verify him (except super-user, not yet merged)

### PR DESCRIPTION
**Pull request by freignat91**
Closes #1291 

Fix the SignUp error which didn't delete the user when internal error happened (as email not working) 

# test

- Do not install the SendGrid key 
- amp -s localhost cluster create -t local --registration=email --notifications=true
- amp -s localhost user signUp -> error SendGrid API key is empty
- amp -s localhost user signUp -> error SendGrid API key is empty

needed to signup two times, in order to verify the user as well been deleted at the first signUp error

### CODE REVIEW

File: data/accounts/store.go [hunk 1]
```diff
@@ -23,6 +23,8 @@ const organizationsRootKey = "organizations"
 const superUser = "su"
 const superOrganization = "so"
 
+const SignUpErrorContextValue = "SignUpError"
```
Comment (bquenin) [inline data/accounts/store.go]: Dead code

**Review by bquenin — CHANGES_REQUESTED** [review]

File: data/accounts/store.go (hunk 1 above)
Comment (freignat91) [reply data/accounts/store.go]: oui, effectivement

Comment (freignat91) [reply data/accounts/store.go]: done

**Review by bquenin — DISMISSED** [review]

**Review by bquenin — APPROVED** [review]
LGTM

**Pull request merged**